src/main/java/com/android/volley/AsyncCache.java

```diff
@@ -24,20 +24,50 @@
      */
     public abstract void get(String key, OnGetCompleteCallback callback);
```

Comment:
Why abstract class?

Reply:
Having it be an abstract class gives us more flexibility, since we can add methods without breaking anything. And the android API guidelines says to favor abstract classes for libraries targeting Java 7, unless you need multiple inheritance.